Expensify/App
# Open E.cash to non-developers

### Details
This PR:
1. Removes the page from the sign-in flow where we were taking github usernames
2. Removes login checks that required a github username
3. Remove all the checks for canAccessExpensify which were blocking users without a GH or shared chat report

### Fixed Issues
Fixes 
https://github.com/Expensify/Expensify/issues/149098
https://github.com/Expensify/Expensify/issues/155153

### Tests
1. Create a new account
2. Make sure you are not asked for your GH username
3. Make sure you get a validation email
4. Click the email and make sure it sends you to e.cash (not e.com)
5. Log out and make sure you can log in normally
6. Log out and make sure you can log into an existing account

### Tested On


### Screenshots
![2021-03-11_13-11-53](https://user-images.githubusercontent.com/42391420/110853000-4f1a3e00-8270-11eb-82ca-9b22e1f6b78e.png)
![2021-03-11_09-37-00](https://user-images.githubusercontent.com/42391420/110853008-50e40180-8270-11eb-888e-c751034190f2.png)
